kf_lib_data_ingest/common/datafile_readers.py

```diff
@@ -76,3 +76,38 @@ def read_json_file(filepath_or_buffer, **kwargs):
     """
     kwargs["convert_dates"] = False
     return pandas.read_json(__ascii_wrap(filepath_or_buffer), **kwargs)
+
+
+def read_file(filepath_or_buffer, original_name=None, **kwargs):
+    """
+    Return contents of a data file as a pandas DataFrame. Wraps more specific
+    readers.
+
+    :param filepath_or_buffer: a data file
+    :type filepath_or_buffer: string (path) or file-like object
+    :param **kwargs: See docs for `pandas IO tools`
+    :return: The structured contents of the file
+    :rtype: pandas.Dataframe
+    """
+    if original_name is None:
+        if isinstance(filepath_or_buffer, str):
+            original_name = filepath_or_buffer
+        else:
+            original_name = filepath_or_buffer.name
+
+    if original_name.endswith((".xlsx", ".xls")):
+        read_func = read_excel_file
+    elif original_name.endswith((".tsv", ".csv")):
```

Comment:
Should we add `.txt`? Excel uses this ext when exporting to tab separated value. Also GitHub doesn't allow `.tsv`in file attachments to descriptions of things (issues, PR, etc).

Reply:
> Should we add .txt?

No, because txt doesn't indicate any particular data layout so you can't convert it to a dataframe without some serious additional intervention.

Reply:
>  Also GitHub doesn't allow .tsvin file attachments to descriptions of things (issues, PR, etc).

That sounds like a dumb bug in github